blog-post/blog-post.md

```diff
@@ -0,0 +1,382 @@
+the code type safety.
+
+```
+function transferTokens(from, to, amount) {
+    fromBalance = balances[from];
+    require(fromBalance >= amount);
+    
+    balances[from] = fromBalance - amount;
+    balances[to] = balances[to] + amount;
+    
+    newFromBalance = balances[from];
+    newToBalance = balances[to];
+    
+    transferSuccess = (newFromBalance == fromBalance - amount) && 
+                      (newToBalance == balances[to] + amount);
+}
+```
+
+## Extended example: Classic Solidity vs Core Solidity 
+
+Now, let's consider an extended example: a contract which implements a unified payment 
+processor that handles three different token standards. The complete Classic Solidity 
+implementation for this simple contract can be found [here.](https://gist.github.com/rodrigogribeiro/7cee270702987fe123921fdbbc2f12ff)
+The code starts by defining a `Payment` struct attempts to represent all three token standards, 
+using the `paymentType` field as a runtime discriminator to determine which fields are relevant 
+for each case.
+
+```
+enum PaymentType { NATIVE, ERC20, ERC721 }
+
+struct Payment {
+    PaymentType paymentType;  // Runtime type tag
+    address token;           // Used for ERC20/ERC721, must be zero for Native 
+    address from;            // Sender address  
+    address to;              // Receiver address
+    uint256 amount;          // Used for Native/ERC20, must be 1 for ERC721
+    uint256 tokenId;         // Used for ERC721, must be zero for others
+}
+```
+
+Next, functions `processPayment` and `calculateFee` uses explicit if/else chains 
+to manually check the `paymentType` and route to the appropriate handling logic.
+Also, in order to avoid invalid states, the code use `require` to ensure that 
+each case is dealing with a proper payment state value.
+
+```
+function processPayment(Payment calldata payment) external {
+  if (payment.paymentType == PaymentType.NATIVE) {
+    require(payment.token == address(0), "Native: no token");
+    require(payment.amount > 0, "Native: amount required");
+    require(payment.tokenId == 0, "Native: no tokenId");
+    payable(payment.to).transfer(payment.amount);
+  } else if (payment.paymentType == PaymentType.ERC20) {
+    require(payment.token != address(0), "ERC20: token required");
+    require(payment.amount > 0, "ERC20: amount required");
+    require(payment.tokenId == 0, "ERC20: no tokenId");
+    IERC20(payment.token).transferFrom(payment.from, payment.to, payment.amount);
+  } else if (payment.paymentType == PaymentType.ERC721) {
+    require(payment.token != address(0), "ERC721: token required");
+    require(payment.amount == 1, "ERC721: amount must be 1");
+    require(payment.tokenId > 0, "ERC721: tokenId required");
+    IERC721(payment.token).transferFrom(payment.from, payment.to, payment.tokenId);
+  }
+}
+```
+
+Now, let's turn our attention to the encoding of this example in Core Solidity.
+The complete encoding of this example can be found [here](). 
```

Comment:
Missing link. Is the example ready?

Reply:
BTW, I don't think "encoding" is the right word here. It sounds weird in this context.